File: 프로그래머스/0/120924. 다음에 올 숫자/다음에 올 숫자.js
function solution(common) {
    const diff = common[1] - common[0];
    if (common[2] - common[1] === diff) {
        return common[common.length - 1] + diff;
    }
    const ratio = common[1] / common[0];
    return common[common.length - 1] * ratio;
}

File: 프로그래머스/0/120924. 다음에 올 숫자/다음에 올 숫자.java
class Solution {
    public int solution(int[] common) {
        int diff = common[1] - common[0];
        if (common[2] - common[1] == diff) {
            return common[common.length - 1] + diff;
        }
        int ratio = common[1] / common[0];
        return common[common.length - 1] * ratio;
    }
}
